### PR TITLE
Updates from workshop feedback with AWS

### PR DESCRIPTION
**Pull request by bildungsroman**
These are *most* of the updates from yesterday's feedback. The remaining todos are:
- create a `final` branch in `dotnet-framework-webapi` for participants to compare to and potentially merge
- add a where we started/where we ended graphic to the **Summary** section at the end of section 4 (Caroll knows more about this)

### CODE REVIEW

File: content/40_modernize_the_api/30_modernize_application.md [hunk 1]
```diff
@@ -100,7 +118,7 @@ You'll see a bunch of angry, red errors. As we manually update the code feel fre
             /* DeleteBook() */
             [ResponseType(typeof(Book))] -> [HttpDelete("{id}")]
             ```
-    4. Search and replace `IHttpActionResult` with `ActionResult`, but for `GetBook()` and `DeleteBook()`, where we need to return a Book record, replace it with `ActionResult<Book>`.
+    4. Search and replace `IHttpActionResult` with `ActionResult`, but for `GetBook()` and `PutBook()`, where we need to return a Book record, replace it with `ActionResult<Book>`.
```
Comment (txase) [inline content/40_modernize_the_api/30_modernize_application.md:121]: The `ActionResult<T>` [generic class](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/generic-classes) is used when we return an object of type T from the controller method that should be sent as a response back to the client (e.g. a 200 response with a body). When we don't respond with a body (e.g. a 204 No Content response), we use `ActionResult` as a non-generic class to signify we aren't returning anything to the client.

The `DeleteBook()` method returns with:
```csharp
return Ok(book);
```
and needs to have a return signature of `ActionResult<Book>`.

The `PutBook()` method returns with:
```csharp
return NoContent();
```
and needs to have a return signature of `ActionResult`.

However, I do think `PostBook()` also needs updating with `ActionResult<Book>` on further review...

**Review by txase — CHANGES_REQUESTED** [review]
Looks good once we fix up the `ActionResult<Book>` stuff!

Why the change to the secrets-reference.png image as well?

**Comment by txase** [comment]
I created a [final](https://github.com/stackery/dotnet-framework-webapi/tree/final) branch. After we merge these changes I'll add a small blurb at the end of the code modifications for people to double check their work against [this diff](https://github.com/stackery/dotnet-framework-webapi/commit/fe150ecfbe3c8dc10947b3e33947e54d3054ade2).

File: content/40_modernize_the_api/30_modernize_application.md (hunk 1 above)
Comment (bildungsroman) [reply content/40_modernize_the_api/30_modernize_application.md:121]: Ah, so this was Marina's suggestion that I implemented

**Comment by bildungsroman** [comment]
> Looks good once we fix up the `ActionResult<Book>` stuff!
> 
> Why the change to the secrets-reference.png image as well?

The env params in that image are wrong when using an existing database, which someone (I forget who) pointed out during the workshop, so I updated it. I also realized I need to update database-reference.png as well which I'll do in the next commit.

**Review by txase — APPROVED** [review]
🎉

**Pull request merged**